### PR TITLE
chore(judicial-system): Improved error responses

### PR DESCRIPTION
**Pull request by gudjong**
# Improved error responses

https://app.asana.com/0/1199153462262248/1201515654267416/f

## What

- Redesigns error responses in the judicial system.
- Modifies log levels.
- Uses shared exception filters in backend, api and xrd-api.
- TODO later: replace HttpExceptions with custom ProblemErrors.

## Why

Clearer logs and easier incident analysis.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/nest/problem/src/lib/problem.module.ts [hunk 1]
```diff
@@ -25,6 +26,22 @@ import { ApolloErrorFilter } from './apolloError.filter'
...
+    return {
+      module: ProblemModule,
+      providers: [
+        {
+          provide: PROBLEM_OPTIONS,
+          useValue: options,
+        },
+      ],
+    }
+  }
```
Comment (eirikurn) [inline libs/nest/problem/src/lib/problem.module.ts:46]: 😘

File: libs/nest/problem/src/lib/base-problem.filter.ts [hunk 2]
```diff
@@ -37,6 +47,12 @@ export abstract class BaseProblemFilter implements ExceptionFilter {
     const response = ctx.getResponse<Response>()
     const problem = this.getProblem(error)
 
+    if (problem.status && problem.status < 500) {
+      this.options.logAllErrors && this.logger.info(error)
+    } else {
+      this.logger.error(error)
+    }
```
Comment (eirikurn) [inline libs/nest/problem/src/lib/base-problem.filter.ts:54]:
```suggestion
    if (problem.status && problem.status >= 500) {
      this.logger.error(error)
    } else if (this.options.logAllErrors) {
       this.logger.info(error)
    }
```

Not a fan of `condition && doStuff` for clarity's sake. Also think it makes sense to reverse the condition to not log all problems with `!problem.status` as errors.

Same above.

File: libs/nest/problem/src/lib/problem.module.ts (hunk 1 above)
Comment (eirikurn) [reply libs/nest/problem/src/lib/problem.module.ts:46]: Oh, and can you please add a test for this logic? 🙏

**Review by eirikurn — COMMENTED** [review]
Really nice and clean! Thank you. Just a couple of minor comments.

File: libs/nest/problem/src/lib/base-problem.filter.ts (hunk 2 above)
Comment (gudjong) [reply libs/nest/problem/src/lib/base-problem.filter.ts:54]: Good points - will change.

File: apps/judicial-system/backend/src/app/modules/case/test/requestRulingSignature.spec.ts [hunk 3]
```diff
@@ -69,12 +58,9 @@ describe('CaseController - Request ruling signature', () => {
       then = await givenWhenThen(caseId, user, theCase)
     })
 
-    it('should initiate signing', () => {
-      expect(then.result.statusCode).toBe(201)
-    })
-
     it('should return a control code and a document token', () => {
-      expect(then.result.message).toEqual({
+      console.log(then)
```
Comment (oddsson) [inline apps/judicial-system/backend/src/app/modules/case/test/requestRulingSignature.spec.ts:62]:
```suggestion
```

**Review by oddsson — CHANGES_REQUESTED** [review]

**Review by oddsson — APPROVED** [review]

File: libs/nest/problem/src/lib/base-problem.filter.spec.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+      // Act
+      await request()
+
+      // Assert
+      expect(infoLog).not.toHaveBeenCalled()
+      expect(errorLog).not.toHaveBeenCalled()
+    })
+  })
+
+  describe('default loggging', () => {
```
Comment (eirikurn) [inline libs/nest/problem/src/lib/base-problem.filter.spec.ts:61]:
```suggestion
  describe('with logAllErrors', () => {
```

**Review by eirikurn — COMMENTED** [review]
Looks awesome! Just one minor comment.

**Review by pshomov — APPROVED** [review]
LGTM

File: libs/nest/problem/src/lib/base-problem.filter.spec.ts (hunk 4 above)
Comment (gudjong) [reply libs/nest/problem/src/lib/base-problem.filter.spec.ts:61]: Thank you for spotting this

**Review by eirikurn — APPROVED** [review]

**Pull request merged**